### PR TITLE
Save-to-cache test simple happy flow with no artifacts

### PR DESCRIPTION
**Pull request by itamaro**

### CODE REVIEW

File: yabt/caching_test.py [hunk 1]
```diff
@@ -100,3 +103,62 @@ def test_prebuilt_targets_build_base_image(basic_conf):
...
+    assert isfile(target_json_path)
+    assert isfile(artifacts_json_path)
+    assert isfile(summary_json_path)
+    assert _EXP_UNZIP_JSON == read_file(target_json_path)
+    assert '{}' == read_file(artifacts_json_path)
+    summary = read_file(summary_json_path, True)
+    assert set(('accessed', 'artifacts_hash', 'build_time', 'created',
+                'name', 'test_time')) == set(summary.keys())
+    assert summary['build_time'] == 5.432
+    assert summary['name'] == ':unzip'
```
Comment (dlehavi-resonai) [inline yabt/caching_test.py]: same remark as above

File: yabt/caching_test.py [hunk 2]
```diff
@@ -100,3 +103,62 @@ def test_prebuilt_targets_build_base_image(basic_conf):
...
+        "version": null
+    },
+    "tags": [
+        "apt-installable"
+    ]
+}"""
+
+
+@pytest.mark.usefixtures('in_caching_project')
+def test_save_target_to_cache(basic_conf):
```
Comment (dlehavi-resonai) [inline yabt/caching_test.py]: the target is what just the program unzip ?

Comment (dlehavi-resonai) [reply yabt/caching_test.py]: yes, that's what I meant (just missing a comma there)

File: yabt/caching_test.py [hunk 3]
```diff
@@ -100,3 +103,62 @@ def test_prebuilt_targets_build_base_image(basic_conf):
...
+    cache_dir = join(basic_conf.project_root, 'yabtwork', '.cache')
+    rmtree(cache_dir)
+    basic_conf.targets = [':all-images']
+    build_context = BuildContext(basic_conf)
+    populate_targets_graph(build_context, basic_conf)
+    unzip_target = build_context.targets[':unzip']
```
[the remark this answers is not in the document]
Comment (itamaro) [reply yabt/caching_test.py]: I prefer explicitness here (and the other one too), as these objects are not the same thing

[the remark this answers is not in the document]
Comment (dlehavi-resonai) [reply yabt/caching_test.py]: I don't understand. the verification is that the target got the intended name - right ?

[the remark this answers is not in the document]
Comment (itamaro) [reply yabt/caching_test.py]: on this specific line, there's no verification at all, just retrieving the target object from internal structure.

on other lines, I verify that serialized cache data exists, and field values are as expected (including target name).

[the remark this answers is not in the document]
Comment (itamaro) [reply yabt/caching_test.py]: ok, done

**Pull request merged**